### PR TITLE
docs(project): Windows setup

### LINKED ISSUE #10949: [Bug]: Windows Environment Setup
1) `yarn build` error on Windows:
```
carbon-components: ./tools/copy-vendor-styles.sh: 2: ./tools/copy-vendor-styles.sh:
carbon-components: : not found
carbon-components: ./tools/copy-vendor-styles.sh: 11: ./tools/copy-vendor-styles.sh:
carbon-components: : not found
carbon-components: ./tools/copy-vendor-styles.sh: 12: set: Illegal option -
lerna ERR! yarn run build exited 2 in 'carbon-components'
```

2) GitHub Desktop error when commiting:
```
.husky/pre-commit: line 4: npx: command not found
husky - pre-commit hook exited with code 127 (error)
```

Environment:
- Windows 10
- Node v16.14.0
- WSL (Ubuntu 18.04 LTS)

### PR DESCRIPTION
**Pull request by NashJames**
Partially Fixes #10949

Error 1 was an End of Line Sequence problem. Changing .sh files to use LF, instead of CRLF should fix it.

#### Changelog

**New**

- Added WSL prerequisite, as it it is not listed anywhere obvious
- Added .gitattributes file for line ending fix

#### Testing / Reviewing

- Run on windows

### CODE REVIEW

**Comment by NashJames** [comment]
I have read the DCO document and I hereby sign the DCO.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ installed:
     version manager for Node.
 - Git
 - [Yarn](https://yarnpkg.com/en/docs/install)
+- [WSL](../docs/guides/setup/windows.md) (Windows Only)
```
Comment (dakahn) [inline .github/CONTRIBUTING.md:32]: This seems cool, the only thought i had is that _technically_ you could contribute using Node for Windows and Powershell -- it's just not tested and officially supported by our dev team.

**Review by dakahn — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (NashJames) [reply .github/CONTRIBUTING.md:32]: Cool idea. No reason people can't contribute in their own way, this should just make it a little clearer for those who aren't concerned about installing WSL.

I could include a note about it on the windows specific guide, mentioning it as an unsupported alternative?

**Review by joshblack — APPROVED** [review]
Thanks so much for doing this! 🙏

**Comment by NashJames** [comment]
Added some instructions too on commiting via Windows with the CLI and `--no-verify` flag. Incase anyone needs it in future

**Pull request merged**